File: php/features.php
<?php
declare(strict_types=1);

// FinancialData SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FinancialDataFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FinancialDataBaseFeature();
            case "test":
                return new FinancialDataTestFeature();
            default:
                return new FinancialDataBaseFeature();
        }
    }
}

File: py/features.py
# FinancialData SDK feature factory

from feature.base_feature import FinancialDataBaseFeature
from feature.test_feature import FinancialDataTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FinancialDataBaseFeature(),
        "test": lambda: FinancialDataTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
